docs/installation/upgrading.rst

```diff
@@ -1,31 +1,97 @@
 .. _upgrading-ksql:
 
-Upgrading KSQL
-==============
+Upgrading ksqlDB
+================
 
-Upgrade one KSQL server at a time (i.e. rolling restart). The remaining KSQL servers should have sufficient spare
-capacity to take over temporarily for unavailable, restarting servers.
+Upgrade one server at a time in a "rolling restart". The remaining servers
+should have sufficient spare capacity to take over temporarily for unavailable,
+restarting servers.
+
+For general guidance on upgrading, see
+[Upgrade ksqlDB](https://docs.ksqldb.io/en/latest/operate-and-deploy/installation/upgrading/).
+
+Upgrading to ksqlDB 5.5 from KSQL 5.4
+-------------------------------------
+
+.. important::
+
+   The upgrade from KSQL 5.4 to ksqlDB 5.5 is not a rolling restart. You must
+   shut down all KSQL instances and then start up all ksqlDB instances, so there
+   will be downtime.
+
+#. Prepare a list of SQL statements for migration. One way is to dump the command
+   topic.
+
+   ```bash
+   ./bin/kafka-console-consumer --bootstrap-server localhost:9092 --topic _confluent-ksql-default__command_topic --from-beginning | jq ".statement" 
+   ```
+
+#. List your application's streams, tables, and types. Don't use the EXTENDED
+   option. Go through each list, and find the the last CREATE statement for each
+   stream/table/type name. Order the CREATE statements so they show the correct
+   dependencies.
+   
+#. Make other compatibility-related changes, as shown in the
+   `upgrade guide <https://docs.ksqldb.io/en/latest/operate-and-deploy/installation/upgrading/#how-to-upgrade>`__.
+
+#. Stop the ``ksql`` service.
+
+#. Uninstall the ksql 5.4 DEB or RPM package.
+
+#. Install the ksqldb 5.5 DEB or RPM package.
+
+#. Copy the version 5.4 configuration file to the ksqldb 5.5 configuration
+   folder, which is located at ``${CONFLUENT_HOME}/etc/ksqldb``:
+
+   ```bash
+   cp /etc/ksql/ksql-server.properties /etc/ksqldb/ksql-server.properties
+   ```
+
+#. Change the ``ksql.service.id`` in the property file, for example, from
+   ``default_`` to ``ksqldb_``.
+
+#. Set up security for your ksqlDB app. ksqlDB supports using
+   :ref:`role-based access control (RBAC) <ksql-rbac>`,
+   :ref:`ACLs <config-security-ksql-acl>`, and no authorization.
+
+#. Create three new role bindings or assign ACLs for the ``ksql`` service
+   principal:
+
+   - Topic: ``__consumer_offsets``
+   - Topic: ``__transaction_state``
+   - TransactionalId: ``ksqldb_``. Use the value that you set in the
+     configuration file.
```

Comment:
Should this be a sub-section of the previous step in case the user has RBAC or ACLs enabled? If not, then this step is not essential for users without security.

Also, what permissions des `ksql` need? write, read, describe?

Reply:
@spena @stan-confluent Are these new role bindings for 5.5? I don't have them documented. @stan-confluent , can you post the rbac commands that you used?

Reply:
Sorry, I don't have the commands - did it via the UI. I granted the ResourceOwner permissions, but I think DeveloperWrite (+ possibly DeveloperManager) would work too - here's the source of truth:
https://github.com/confluentinc/ksql/pull/3660/files#diff-b2344446085218073ba234df216f085d
You can see which roles would grant you these permissions here - https://github.com/confluentinc/ce-kafka/blob/master/ce-rbac/src/main/resources/default_rbac_roles.json

Reply:
@stan-confluent @spena I added the ACLs and role bindings, let me know if they're in the ballpark.